core/Config/IniFileChain.php

```diff
@@ -0,0 +1,342 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Config;
+
+use Piwik\Ini\IniReader;
+use Piwik\Ini\IniReadingException;
+use Piwik\Ini\IniWriter;
+use Piwik\Piwik;
+
+/**
+ * Manages a list of INI files where the settings in each INI file merge with or override the
+ * settings in the previous INI file.
+ *
+ * The IniFileChain class manages two types of INI files: multiple default setting files and one
+ * user settings file.
+ *
+ * The default setting files (for example, global.ini.php & common.ini.php) hold the default setting values.
+ * The settings in these files are merged recursively, so array settings in one file will add elements to
+ * the same settings in the previous file. Default settings files cannot be modified through the IniFileChain
+ * class.
+ *
+ * The user settings file (for example, config.ini.php) holds the actual setting values. Settings in the
+ * user settings files overwrite other settings. So array settings will not merge w/ previous values.
```

Comment:
This couldn't have been better explained. I guess almost nobody knew it worked this way - good on you for making this clear and better unit tested!